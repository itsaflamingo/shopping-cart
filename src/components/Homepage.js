import React, {useState} from 'react'
import {useNavigate} from 'react-router-dom';
import Carousel from './Carousel'

function Homepage() {
  const nav = useNavigate();

  const navToShop = () => {
    nav('/shop')
  }

  return (
    <div className="App">
          <Carousel navToShop={navToShop} />
    </div>
  );
}

export default Homepage;
