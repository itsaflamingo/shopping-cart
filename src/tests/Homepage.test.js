import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";  // optional
import Homepage from '../components/Homepage'
import {BrowserRouter} from 'react-router-dom';
import { act } from "react-dom/test-utils";

describe('Homepage component', () => {
    it('Renders correctly', () => {
        const {container} = render(
            <BrowserRouter>
                <Homepage />
            </BrowserRouter>
        )
        expect(container).toMatchSnapshot()
    })
    it('renders correct heading', () => {
        render(
        <BrowserRouter>
            <Homepage />
        </BrowserRouter>)
        expect(screen.getByRole('heading').textContent).toMatch(/Fake Shop/i)
    })

    it('page switches to shop on button click', () => {
        render(
            <BrowserRouter>
                <Homepage />
            </BrowserRouter>)

        const button = screen.getByRole('button', {name: 'Shop'});
        act(() => {
            button.dispatchEvent(new MouseEvent('click', {bubbles: true}))
        });
        expect(window.location.pathname).toEqual('/shop');
    })
})