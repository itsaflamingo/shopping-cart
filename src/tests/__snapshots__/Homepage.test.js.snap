// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component Renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <div
      id="background"
      style="background-image: url(background.jpg); background-size: cover; background-repeat: no-repeat;"
    >
      <div
        id="home-header"
      >
        <h2>
          Fake Shop
        </h2>
      </div>
      <button
        id="visit-shop"
      >
        Shop
      </button>
    </div>
  </div>
</div>
`;
